test/api/route/messages_test.dart

```diff
@@ -3,16 +3,177 @@ import 'dart:convert';
 import 'package:checks/checks.dart';
 import 'package:http/http.dart' as http;
 import 'package:test/scaffolding.dart';
+import 'package:zulip/api/model/model.dart';
 import 'package:zulip/api/model/narrow.dart';
 import 'package:zulip/api/route/messages.dart';
 import 'package:zulip/model/narrow.dart';
 
 import '../../example_data.dart' as eg;
 import '../../stdlib_checks.dart';
 import '../fake_api.dart';
+import '../model/model_checks.dart';
 import 'route_checks.dart';
 
 void main() {
+  group('getMessageCompat', () {
+    Future<Message?> checkGetMessageCompat(FakeApiConnection connection, bool expectLegacy, {
+      required int messageId,
+      bool? applyMarkdown,
+    }) async {
+      final result = await getMessageCompat(connection,
+        messageId: messageId,
+        applyMarkdown: applyMarkdown,
+      );
+      expectLegacy
+        ? (check(connection.lastRequest).isA<http.Request>()
```

Comment:
Seems a bit cleaner to make this a normal if/else statement, since we're not using the expression nature of the `?:` operator.